components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -208,6 +213,205 @@ private OAuth2AuthorizeRespDTO validateAuthzRequest(OAuth2AuthorizeReqDTO authzR
         return authorizeRespDTO;
     }
 
+    /**
+     * Handle authorization request before consent page.
+     *
+     * @param authzReqDTO authzReqDTO
+     * @return OAuthAuthzReqMessageContext
+     */
+    public OAuthAuthzReqMessageContext handleAuthorizationBeforeConsent(OAuth2AuthorizeReqDTO authzReqDTO)
+            throws IdentityOAuth2Exception, InvalidOAuthClientException, IdentityOAuth2ScopeValidationException {
+
+        OAuthAuthzReqMessageContext authzReqMsgCtx = getOAuthAuthzReqMessageContext(authzReqDTO);
+        ResponseTypeHandler authzHandler = getResponseHandler(authzReqDTO);
+        //validate requested scopes
+        validateRequestedScopes(authzReqMsgCtx, authzHandler);
+        return authzReqMsgCtx;
+    }
+
+    /**
+     * Handle authorization request after consent page.
+     *
+     * @param authzReqMsgCtx authzReqMsgCtx
+     * @return OAuth2AuthorizeRespDTO
+     */
+    public OAuth2AuthorizeRespDTO handleAuthorizationAfterConsent(OAuthAuthzReqMessageContext authzReqMsgCtx)
+            throws IdentityOAuth2Exception {
+
+        OAuth2AuthorizeReqDTO authzReqDTO =  authzReqMsgCtx.getAuthorizationReqDTO();
+        ResponseTypeHandler authzHandler = getResponseHandler(authzReqDTO);
+
+        OAuth2AuthorizeRespDTO authorizeRespDTO = new OAuth2AuthorizeRespDTO();
+        if (isInvalidClient(authzReqDTO, authorizeRespDTO, authzReqMsgCtx, authzHandler)) {
+            return authorizeRespDTO;
```

Comment:
let's add a log here

Reply:
there is a log in the isInvalidClient method for that